lib/Sema/CSGen.cpp

```diff
@@ -2368,7 +2369,8 @@ namespace {
               CS.getConstraintLocator(locator), TVO_CanBindToNoEscape);
```

Comment:
I think in this case we should have locator with points to `pattern` and ends with `OptionalPayload`.